framework/wazuh/core/indexer/agent.py

```diff
@@ -14,41 +17,44 @@ class AgentsIndex(BaseIndex):
     INDEX = 'agents'
     SECONDARY_INDEXES = []
 
-    async def create(self, id: str, key: str, name: str) -> Agent:
+    async def create(self, id: str, key: str, name: str, groups: str = None) -> dict:
         """Create a new agent.
 
         Parameters
         ----------
         id : str
-            Identifier of the new agent.
-        key : str
-            Key of the new agent.
+            New agent ID.
         name : str
-            Name of the new agent.
-
-        Returns
-        -------
-        Agent
-            The created agent instance.
+            New agent name.
+        key : str
+            New agent key.
+        groups : str
+            New agent groups.
 
         Raises
         ------
         WazuhError(1708)
             When already exists an agent with the provided id.
+
+        Returns
+        -------
+        agent_dict : dict
+            The created agent instance in a dictionary.
         """
-        agent = Agent(id=id, raw_key=key, name=name)
+        agent = Agent(raw_key=key, name=name, groups='default' + f',{groups}' if groups else '')
```

Comment:
Agents are always assigned the `default` group on creation

Reply:
This is something that we need yet to define. I'd keep it

Reply:
I will ask in the persistence channel to confirm before making the changes because this affects how we handle agent's information in general

Reply:
Leaving it like this until we get a response, in any case we could resolve this in the issue https://github.com/wazuh/wazuh/issues/25592